### PR TITLE
Smarter formatting based on amount of data

### PR DESCRIPTION
**Pull request by sulkaharo**
To make sure the rendering works for both SMB and non-SMB users.

@scottleibrand can you test if this still works as you intended for SMB users?

### CODE REVIEW

File: lib/client/renderer.js [hunk 1]
```diff
@@ -444,13 +444,17 @@ function init (client, d3) {
...
+      
       var unit_of_measurement = ' U'; // One international unit of insulin (1 IU) is shown as '1 U'
-      if ( treatment.insulin < 1 && !treatment.carbs ) { // don't show the unit of measurement for insulin boluses < 1 without carbs (e.g. oref0 SMB's). Otherwise lot's of small insulin only dosages are often unreadable
+      var enteredBy = '' + treatment.enteredBy;
+      
+      if ( treatment.insulin < 1 && !treatment.carbs && enteredBy.indexOf('openaps') > -1) { // don't show the unit of measurement for insulin boluses < 1 without carbs (e.g. oref0 SMB's). Otherwise lot's of small insulin only dosages are often unreadable
          unit_of_measurement = '';
+         dosage_units = (dosage_units+"").replace(/^0/,"");
       }
       // remove leading zeros to avoid overlap with adjacent boluses
```
Comment (scottleibrand) [inline lib/client/renderer.js]: Comment should move up too.

**Comment by scottleibrand** [comment]
Is this counting the number of treatments on-screen?  Looks like the result is a larger font size for our typical number of visible treatments, but still readable.  Might be slightly better to increase the spacing between the circle and the text, but also maybe not worth worrying about.

**Comment by sulkaharo** [comment]
how many treatments / day does SMB create for you / day on average?

**Comment by scottleibrand** [comment]
Looks like we're averaging about 150 SMBs per day with no-bolus when the sensor is working well.  Only half that in the last 24h due to jumpy sensor and resulting less frequent larger SMBs.

**Comment by sulkaharo** [comment]
Right, with those amounts this should produce exactly the same font size as your original change, so odd if you’re seeing something different.

**Comment by scottleibrand** [comment]
Is it based on the number of treatments per day?  Or the number on-screen?

**Comment by scottleibrand** [comment]
It looks like it's using the number on-screen, and reducing the font size as I shift from the 2h and 4h to the 12h and 24h graphs and it has to fit more treatments on-screen.  If that's the desired behavior, I think that's fine.

**Pull request merged**